consensus/p2p/buffered/buffered_topic_subscription.go

```diff
@@ -0,0 +1,44 @@
+package buffered
+
+import (
+	"context"
+	"errors"
+
+	"github.com/NethermindEth/juno/utils"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+type TopicSubscription struct {
+	log        utils.Logger
+	bufferSize int
+	callback   func(context.Context, *pubsub.Message)
+}
+
+func New(log utils.Logger, bufferSize int, callback func(context.Context, *pubsub.Message)) TopicSubscription {
```

Comment:
nitpick, shouldn't `New` be called `NewTopicSubscription`, thinking the use of this type outside the package where we will do

```
x := buffered.New(...)
```

to 

```
x := buffered.NewTopicSubscription(...)
```
If this is the only type meant to be inside the package, I think it's ok to leave the original constructor name `New`